test/acceptance/asset-discovery.test.js

```diff
@@ -0,0 +1,27 @@
+import expect from 'expect'
+import {
+  run,
+  setupApiProxy,
+  setupDummyApp
+} from './helpers'
+
+describe('Asset discovery', () => {
+  let proxy = setupApiProxy()
+  let dummy = setupDummyApp()
+
+  it('does not create snapshots while stopping a build', async () => {
```

Comment:
More accurately: `waits for snapshot finalization before build finalization`

Reply:
Snapshot creation*? I guess technically we want to make sure both creation and finalization happen before the build finalizes

Reply:
I like it 👍 We don't _have_ to finalize the snapshot right now (the API doesn't require it). But it is nice to do that for perf